NPM cannot resolve Astro Actions virtual module

### Astro Info

```block
I'm working on a WSL2 setup, but here's the info I got from `astro info`:

Astro                    v5.14.6
Vite                     v6.4.0
Node                     v22.20.0
System                   Linux (x64)
Package Manager          npm
Output                   server
Adapter                  @astrojs/node (v9.5.0)
```

### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

I've been developing my project with Astro using `pnpm`, but when I try running my project with `npm`, I get this error when I start my development server using `npm run dev`:

```bash
✘ [ERROR] Could not resolve "virtual:astro:actions/options"

    node_modules/astro/dist/virtual-modules/actions.js:1:42:
      1 │ import { shouldAppendTrailingSlash } from "virtual:astro:actions/options";
        ╵                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  You can mark the path "virtual:astro:actions/options" as external to exclude it from the bundle,
  which will remove this error and leave the unresolved path in the bundle.

✘ [ERROR] Could not resolve "virtual:astro:actions/runtime"

    node_modules/astro/dist/virtual-modules/actions.js:10:14:
      10 │ export * from "virtual:astro:actions/runtime";
         ╵               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  You can mark the path "virtual:astro:actions/runtime" as external to exclude it from the bundle,
  which will remove this error and leave the unresolved path in the bundle.

18:52:47 [ERROR] [UnhandledRejection] Astro detected an unhandled rejection. Here's the stack trace:
Error: Build failed with 2 errors:
node_modules/astro/dist/virtual-modules/actions.js:1:42: ERROR: Could not resolve "virtual:astro:actions/options"
node_modules/astro/dist/virtual-modules/actions.js:10:14: ERROR: Could not resolve "virtual:astro:actions/runtime"
```

I can't find any way where this unhandled rejection might be coming from and prevents a dev server from running at all.

### What's the expected result?

I would expect a more precise message of _where_ this unhandled error would be instead of throwing an error immediately on a virtual import.

I'll link to the project this issue is happening at the moment since producing a minimal reproduction didn't seem to throw the same error.

### Link to Minimal Reproducible Example

https://github.com/haetae-bit/fanfic-atproto

### Participation

- [ ] I am willing to submit a pull request for this issue.

Experiencing the same error as OP:

```bash
09:46:13 [ERROR] [UnhandledRejection] Astro detected an unhandled rejection. Here's the stack trace:
Error: Build failed with 2 errors:
node_modules/astro/dist/virtual-modules/actions.js:1:42: ERROR: Could not resolve "virtual:astro:actions/options"
node_modules/astro/dist/virtual-modules/actions.js:10:14: ERROR: Could not resolve "virtual:astro:actions/runtime"
    at failureErrorWithLog (F:\GitHub\newgentdigital\website\node_modules\vite\node_modules\esbuild\lib\main.js:1467:15)
    at F:\GitHub\newgentdigital\website\node_modules\vite\node_modules\esbuild\lib\main.js:926:25
    at F:\GitHub\newgentdigital\website\node_modules\vite\node_modules\esbuild\lib\main.js:1345:9
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
  Hint:
    Make sure your promises all have an `await` or a `.catch()` handler.
  Error reference:
    https://docs.astro.build/en/reference/errors/unhandled-rejection/
  Stack trace:
    at F:\GitHub\newgentdigital\website\node_modules\astro\dist\virtual-modules\actions.js: ERROR: Could not resolve "virtual:astro:actions\options":1:42
    [...] See full stack trace in the browser, or rerun with --verbose
```

As suggested by @florian-lefebvre - downgrading to 5.14.5 solves this issue. Building the project works fine, development mode does not.

For (a not so minimal) reproduction, clone https://github.com/newgentdigital/website and run `bun dev` or `astro dev` as is.

My `astro info` output:

```
Astro                    v5.14.6
Node                     v22.20.0
System                   Windows (x64)
Package Manager          bun
Output                   static
Adapter                  @astrojs/cloudflare
Integrations             @astrojs/sitemap
                         @astrojs/mdx
                         @sentry/astro
                         @astrojs/react
```

--- a/.changeset/calm-bottles-bathe.md
+++ b/.changeset/calm-bottles-bathe.md
@@ -1,0 +1,5 @@
+---
+'astro': patch
+---
+
+Fixes a regression that caused Actions to throw errors while loading

--- a/packages/astro/src/actions/consts.ts
+++ b/packages/astro/src/actions/consts.ts
@@ -1,6 +1,7 @@
 export const ACTIONS_TYPES_FILE = 'actions.d.ts';
 
 export const VIRTUAL_MODULE_ID = 'astro:actions';
+export const RESOLVED_VIRTUAL_MODULE_ID = '\0' + VIRTUAL_MODULE_ID;
 
 /** Used to expose shared utilities, with server or client specific implementations */
 export const RUNTIME_VIRTUAL_MODULE_ID = 'virtual:astro:actions/runtime';

--- a/packages/astro/src/actions/runtime/virtual.ts
+++ b/packages/astro/src/actions/runtime/virtual.ts
@@ -1,5 +1,6 @@
 import { shouldAppendTrailingSlash } from 'virtual:astro:actions/options';
-import type { ActionClient, SafeResult } from '../actions/runtime/server.js';
+import type { APIContext } from '../../types/public/context.js';
+import type { ActionClient, SafeResult } from './server.js';
 import {
 	ACTION_QUERY_PARAMS,
 	ActionError,
@@ -7,8 +8,7 @@ import {
 	astroCalledServerError,
 	deserializeActionResult,
 	getActionQueryString,
-} from '../actions/runtime/shared.js';
-import type { APIContext } from '../types/public/context.js';
+} from './shared.js';
 
 export * from 'virtual:astro:actions/runtime';
 

--- a/packages/astro/src/actions/vite-plugin-actions.ts
+++ b/packages/astro/src/actions/vite-plugin-actions.ts
@@ -13,6 +13,7 @@ import {
 	RESOLVED_NOOP_ENTRYPOINT_VIRTUAL_MODULE_ID,
 	RESOLVED_OPTIONS_VIRTUAL_MODULE_ID,
 	RESOLVED_RUNTIME_VIRTUAL_MODULE_ID,
+	RESOLVED_VIRTUAL_MODULE_ID,
 	RUNTIME_VIRTUAL_MODULE_ID,
 	VIRTUAL_MODULE_ID,
 } from './consts.js';
@@ -62,7 +63,7 @@ export function vitePluginActions({
 		enforce: 'pre',
 		async resolveId(id) {
 			if (id === VIRTUAL_MODULE_ID) {
-				return this.resolve('astro/virtual-modules/actions.js');
+				return RESOLVED_VIRTUAL_MODULE_ID
 			}
 
 			if (id === RUNTIME_VIRTUAL_MODULE_ID) {
@@ -99,6 +100,10 @@ export function vitePluginActions({
 			server.watcher.on('change', watcherCallback);
 		},
 		async load(id, opts) {
+			if (id === RESOLVED_VIRTUAL_MODULE_ID) {
+				return { code: `export * from 'astro/actions/runtime/virtual.js';` }
+			}
+
 			if (id === RESOLVED_NOOP_ENTRYPOINT_VIRTUAL_MODULE_ID) {
 				return { code: 'export const server = {}' };
 			}

